core/include/log.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @addtogroup  core_util
+ * @{
+ *
+ * @file        log.h
+ * @brief       System logging header
+ *
+ * This header offers a bunch of "LOG_*" functions that, with the default
+ * implementation, behave like printf with verbosity levels.
+ *
+ * If desired, it is possible to implement a log module which then will be used
+ * instead the default printf-based implementation.  In order to do so, the log
+ * module has to provide "log_module.h", add -DHAVE_LOG_MODULE to CFLAGS and
```

Comment:
Hm, I don't like that the module has to modify the CFLAGS somehow. How about defining a PSEUDOMODULE instead?